src/ResourceManager/Resources/Commands.Resources/Models.ResourceGroups/ResourceClient.cs

```diff
@@ -325,16 +325,17 @@ public static List<string> ParseDetailErrorMessage(string statusMessage)
             params string[] status)
         {
             DeploymentExtended deployment;
+            Delay stepedDelay = new Delay(firstDelay: 0);
 
             do
             {
+                TestMockSupport.Delay(stepedDelay.GetNextDelay());
```

Comment:
Why can't we just do:
int counter = 0;
do
{
  counter =  counter + 5000 > 60000 ? 60000 : counter + 5000;
  TestMockSupport.Delay(counter);
...
}